evalml/automl/automl_search.py

```diff
@@ -896,4 +930,31 @@ def score_pipelines(self, pipelines, X_holdout, y_holdout, objectives):
             Note that the any pipelines that error out during scoring will not be included in the dictionary
             but the exception and stacktrace will be displayed in the log.
         """
-        return self._engine.score_batch(pipelines, X_holdout, y_holdout, objectives)
+        scores = {}
+        objectives = [get_objective(o, return_instance=True) for o in objectives]
+
+        computations = []
+        for pipeline in pipelines:
+            computations.append(self._engine.submit_scoring_job(self.automl_data, pipeline, X_holdout, y_holdout, objectives))
+
+        while computations:
+            computation = computations.pop(0)
+            if computation.done():
+                pipeline_name = computation.meta_data["pipeline_name"]
+                try:
+                    scores[pipeline_name] = computation.get_result()
```

Comment:
It's not exactly the same because the results are not the same between the training job and the scoring job. So when we call `get_result()` in the scoring job we expect to just get back a dictionary of scores as opposed to a pipeline class. 

I guess we could modify the scoring job to return the pipeline that was scored as well that would side-step the need for the metadata.